crates/vm/src/wasm/run.rs

```diff
@@ -519,6 +518,65 @@ where
     pub cache_access: PhantomData<*const CA>,
 }
 
+impl<'a, S, CA> namada_vp::native_vp::VpEvaluator<'a, S, VpCache<CA>, Self>
+    for VpEvalWasm<<S as StateRead>::D, <S as StateRead>::H, CA>
+where
+    S: 'static + StateRead,
+    CA: WasmCacheAccess,
+{
+    fn eval(
+        ctx: &namada_vp::native_vp::Ctx<'a, S, VpCache<CA>, Self>,
+        vp_code_hash: Hash,
+        input_data: BatchedTxRef<'_>,
+    ) -> namada_state::StorageResult<()> {
+        use std::marker::PhantomData;
+
+        use namada_state::ResultExt;
+
+        use crate::host_env::VpCtx;
+        use crate::wasm::run::VpEvalWasm;
+
+        let eval_runner =
+            VpEvalWasm::<<S as StateRead>::D, <S as StateRead>::H, CA> {
+                db: PhantomData,
+                hasher: PhantomData,
+                cache_access: PhantomData,
+            };
+        let mut iterators: PrefixIterators<'_, <S as StateRead>::D> =
+            PrefixIterators::default();
+        let mut result_buffer: Option<Vec<u8>> = None;
+        let mut yielded_value: Option<Vec<u8>> = None;
+        let mut vp_wasm_cache = ctx.vp_wasm_cache.clone();
+
+        let ctx = VpCtx::new(
+            ctx.address,
+            ctx.state.write_log(),
+            ctx.state.in_mem(),
+            ctx.state.db(),
+            ctx.gas_meter,
+            ctx.tx,
+            ctx.cmt,
+            ctx.tx_index,
+            &mut iterators,
+            ctx.verifiers,
+            &mut result_buffer,
+            &mut yielded_value,
+            ctx.keys_changed,
+            &eval_runner,
+            &mut vp_wasm_cache,
+        );
+        eval_runner
+            .eval_native_result(ctx, vp_code_hash, input_data)
+            .inspect_err(|err| {
+                tracing::warn!(
+                    "VP eval from a native VP failed with:
+            {err}",
+                );
```

Comment:
nit: fix this weird formatting

Reply:
8ca5943e1530c65bd1d78c43f8db728f5bc77318